### PR TITLE
[Merged by Bors] - Separate committee subscriptions queue

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

As we've seen on Prater, there seems to be a correlation between these messages

```
WARN Not enough time for a discovery search  subnet_id: ExactSubnet { subnet_id: SubnetId(19), slot: Slot(3742336) }, service: attestation_service
```

... and nodes falling 20-30 slots behind the head for short periods. These nodes are running ~20k Prater validators.

After running some metrics, I can see that the `network_recv` channel is processing ~250k `AttestationSubscribe` messages per minute. It occurred to me that perhaps the `AttestationSubscribe` messages are "washing out" the `SendRequest` and `SendResponse` messages. In this PR I separate the `AttestationSubscribe` and `SyncCommitteeSubscribe` messages into their own queue so the `tokio::select!` in the `NetworkService` can still process the other messages in the `network_recv` channel without necessarily having to clear all the subscription messages first.

~~I've also added filter to the HTTP API to prevent duplicate subscriptions going to the network service.~~

## Additional Info

- Currently being tested on Prater

### CODE REVIEW

**Comment by paulhauner** [comment]
This PR *temporarily* added a cache to the BN HTTP API to filter duplicate subscriptions before they were sent to the network service (removed in [e4e4152](https://github.com/sigp/lighthouse/pull/3508/commits/e4e41522a3da44336e747fadab29d6292ddc661c)). However, after some discussion with @divagant-martian, it was decided that the network service relies on those duplicates in some capacity and the cache is likely to introduce problems.

I've removed the cache accordingly. On our Prater nodes that were struggling, the cache caused them to only log some `Unable to process committee subscriptions` ("full queue") messages during startup but then run fine afterwards. *Without* the cache, some nodes continue to log the "full queue" messages every 1-2 epochs. However, those nodes still follow the head perfectly fine.

In summary, this PR does not solve the core issue of the BN lagging to expediently process subnet subscriptions with 20k+ validators. However it does isolate that problem so that it does not prevent the node from following the head. Overall, I think this PR is a big improvement but more work is required.

File: beacon_node/http_api/src/metrics.rs [hunk 1]
```diff
@@ -41,4 +41,9 @@ lazy_static::lazy_static! {
         "http_api_block_published_very_late_total",
         "The count of times a block was published beyond the attestation deadline"
     );
+    pub static ref HTTP_API_SUBSCRIPTION_SEEN_CACHE: Result<IntCounterVec> = try_create_int_counter_vec(
+        "http_api_subscription_seen_cache",
+        "Tracks the `ValidatorSubscriptionSeenCache`",
+        &["status"]
+    );
```
Comment (michaelsproul) [inline beacon_node/http_api/src/metrics.rs]: Can delete this

**Review by michaelsproul — APPROVED** [review]
Looks good modulo one smol change!

**Comment by michaelsproul** [comment]
bors retry
